src/main/perl/FileEditor.pm

```diff
@@ -58,13 +66,45 @@ sub _is_valid_source
     return -f $fn;
 }
 
+sub _is_reference_newer
+{
+    my ($self) = @_;
+    my $is_newer = 0;   # Assume false
+    if (  exists(*$self->{options}->{source}) && $self->_is_valid_source(*$self->{options}->{source}) ) {
+        # stat()[9] is modification time
+        if ( !$self->_is_valid_source(*$self->{filename}) || 
+             ((stat(*$self->{options}->{source}))[9] > (stat(*$self->{filename}))[9]) ) {
+          $is_newer = 1
+        }
+    }
+    #FIXME: replace by $self->debug() after PR #154 has been merged...
+    if ( *$self->{LOG} ) {
+        if ( $is_newer ) {
+            # As this is a non reproducible event, be sure to log it when it happens
+            *$self->{LOG}->info("File ", *$self->{filename}, " contents reset to reference file (", *$self->{options}->{source}, ") contents.");
+        } else {
+            *$self->{LOG}->debug(1, "Reference file (", *$self->{options}->{source}, ") older than ", *$self->{filename}, ": ignoring it");
+        };
+    };
+    return $is_newer;
+}
 
 sub new
 {
     my $class = shift;
     my $self = $class->SUPER::new (@_);
-    if ($self->_is_valid_source(*$self->{filename})) {
-        my $txt = LC::File::file_contents (*$self->{filename});
+    my $src_file;
+    my ($path, %opts) = @_;
+
+    *$self->{options}->{source} = $opts{source} if exists ($opts{source});
+    if ( $self->_is_reference_newer() ) {
+        $src_file = *$self->{options}->{source};
+    } elsif ($self->_is_valid_source(*$self->{filename})) {
```

Comment:
@stdweird in fact I am not sure to completely understand your point. In the previous code, nothing prevented filename to be a directory for example. Just it was not read but when you were closing the file, you would have got an error. I don't think the new code is really different. But it may be good to add the test you suggest, i.e. be sure that the file wil be writable at the end. May be better to handle as a separate issue...